### ISSUE TITLE
Enhance REST API to easier identify latest app version

### ISSUE THREAD

**Issue by jurgenhaas** [nextcloud/appstore#565]
The endpoint [GET /api/v1/platform/{platform-version}/apps.json](https://nextcloudappstore.readthedocs.io/en/latest/restapi.html#get-all-apps-and-releases) currently provides something like this for each app:

```
[
    {
        "id": "name",
        "releases": [
            {
                "version": "1.0.0",
            },
            {
                "version": "1.0.1",
            }
        ]
    }
]
```

This way it is difficult to identify the latest available release. In fact you have to iterate through all releases and do a version number comparison over the version value which is not simple and makes it impossible for apps that have no such logic like e.g. Ansible or other provisioning tools.

I'm asking to enhance the API such that we provide a latest release value and then also index the list by the version keys:

```
[
    {
        "id": "name",
        "latest": "1.0.1",
        "releases": {
            "1.0.0": {
                "version": "1.0.0",
            },
            "1.0.1": {
                "version": "1.0.1",
            }
        }
    }
]
```

Is that something that you are willing to support?

**Comment by BernhardPosselt** [comment]
The API was designed to be easily cacheable so that you can replace the whole store with a directory of JSON files. Therefore filters were explicitely only implemented for nextcloud versions.

Ansible should be scriptable with Python right? Just install the same lib (semantic_version) in your script. 

Need to think about this some more, right now I'm busy :)

**Comment by jurgenhaas** [comment]
Yes, Ansible is scriptable with Python but so far I don't have to write a plugin to do all that's needed to install Nextcloud and addons. I'm reading the json from the REST API and grab the first release in the release list assuming that's the latest. That worked until the plugin `richdocuments` released version 2.0.10 which is no longer the first one in the list but somewhere in the middle close to 2.0.1 which ios because the sorting works alphabetically and not according to semantic versioning.

So, instead of my original proposal what would also make things much cleaner is to sort the release list semantically and not alphabetically. Maybe that's easier and wouldn't break the API.

**Comment by BernhardPosselt** [comment]
Right sorting semantically is probably a very good idea and would fix your particular issue. Currently not sure however how we should deal with nightlies and pre-releases

**Comment by jurgenhaas** [comment]
Good point, maybe putting them into a separate list?

**Comment by BernhardPosselt** [comment]
Probably not :) that would make things really complicated. However putting the latest stable version into a separate attribute is probably a good idea.

**Comment by jurgenhaas** [comment]
Sure, but we should also have a chance to access all the details of the latest stable release directly without preprocessing anything.

**Comment by BernhardPosselt** [comment]
Ah right, you are correct.

**Comment by BernhardPosselt** [comment]
Maybe @nickvergessen @MorrisJobke  @juliushaertl  have an idea that could make sense for the server as well.
